### PR TITLE
feat: Implement dying

### PR DESCRIPTION
**Pull request by meczajkowski**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
Tamagotchi should die when health is below 0

### Requirements
1. Should set `dead` state when health is below 0
2. Action buttons should be disabled, and replaced with Restart button

### Code Review

The provided code changes seem to handle the UI updates when the Tamagotchi's state is 'dead'. It correctly disables the game action buttons and enables the game over (restart) button. However, there is no code change that sets the Tamagotchi's state to 'dead' when its health is below 0.

The code changes do not fully cover the task requirements. The first requirement is not met in the provided code changes. The code that sets the Tamagotchi's state to 'dead' when its health falls below 0 is missing. 

The second requirement is met. The action buttons are correctly disabled and replaced with a Restart button when the Tamagotchi's state is 'dead'. 

In conclusion, additional code changes are needed to meet all task requirements. Specifically, the condition to set the Tamagotchi's state to 'dead' when its health falls below 0 needs to be implemented.

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task goal
Tamagotchi should die when health is below 0

### Requirements
1. Should set `dead` state when health is below 0
2. Action buttons should be disabled, and replaced with Restart button

### Code changes review

The provided code changes seem to cover the task requirements.

1. The `setState` method has been updated to set the Tamagotchi's state to 'dead' when its health falls below 0. This covers the first requirement.

```diff
@@ -122,6 +139,7 @@ export default class Tamagotchi {
 
   setState() {
     if (this.health.value <= 0) {
+      // set the Tamagotchi's state to 'dead' when its health falls below 0
       this.state = 'dead';
     } else if (this.isEating) {
       this.state = 'eating';
```

2. The `updateDisplay` method has been updated to disable the action buttons and replace them with a Restart button when the Tamagotchi's state is 'dead'. This covers the second requirement.

```diff
@@ -58,10 +58,27 @@ export default class Tamagotchi {
   };
 
   updateDisplay() {
+    const buttonsGameon = document.querySelector('.action-buttons__gameon');
+    const buttonsGameover = document.querySelector('.action-buttons__gameover');
+
     this.displayHealth('.health');
     this.displayHunger('.hunger');
     this.displayEnergy('.energy');
     this.displayFun('.fun');
+
+    if (this.state === 'dead') {
+      buttonsGameon.classList.remove('active');
+      buttonsGameon.setAttribute('disabled', true);
+
+      buttonsGameover.classList.add('active');
+      buttonsGameover.setAttribute('disabled', false);
+    } else {
+      buttonsGameon.classList = 'action-buttons__gameon active';
+      buttonsGameon.setAttribute('disabled', false);
+
+      buttonsGameover.classList = 'action-buttons__gameover';
+      buttonsGameover.setAttribute('disabled', true);
+    }
   }
```

Overall, the code changes are in line with the task requirements.
